### DISCUSSION TITLE
Building the phar file

### DISCUSSION THREAD

**Discussion by GeneColeman** [wp-cli/wp-cli#6014] [How do I?]
Hello,
I am trying to build this project into a phar file.  I have composer installed but when I run `composer install` I am getting this:
```
Composer could not detect the root package (wp-cli/wp-cli) version, defaulting to '1.0.0'. See https://getcomposer.org/root-version
No composer.lock file present. Updating dependencies to latest instead of installing from lock file. See https://getcomposer.org/install for more information.
Loading composer repositories with package information
Updating dependencies
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - wp-cli/wp-cli is present at version 1.0.0+no-version-set and cannot be modified by Composer
    - Root composer.json requires roave/security-advisories dev-latest -> satisfiable by roave/security-advisories[dev-latest].
    - roave/security-advisories dev-latest conflicts with wp-cli/wp-cli >=0.12,<2.5.
```
So my questions is this - how do I build from start to finish with the current version?

BTW - Yes, I have looked at the wp-cli documentation but there is little there and nothing about the actual building of the phar.

Any help would be appreciated.

**Comment by swissspidy** [comment]
You can check the GitHub Actions workflow in the wp-cli-bundle repo to see how we build the Phar file there:

https://github.com/wp-cli/wp-cli-bundle/blob/a4e0ef11d0e1b45ade6844209333fd1f20082af8/.github/workflows/deployment.yml#L12-L64
